packages/botonic-cli/src/commands/serve.ts

```diff
@@ -11,20 +16,120 @@ export default class Run extends Command {
+      default: true,
+    }),
+  }
 
   static args = []
 
   private telemetry = new Telemetry()
 
   async run(): Promise<void> {
     this.telemetry.trackServe()
-    this.parse(Run)
-    try {
-      console.log(colors.blue('\nServing Botonic...'))
-      spawnNpmScript('start')
-    } catch (e) {
-      console.log(e)
+    const { flags } = this.parse(Run)
+    if (flags.preview) {
+      const apps = ['websocket', 'rest', 'webchat', 'webviews']
+      const BASE_PATH = process.cwd()
+      const API_REST_DIR_SRC = path.resolve(BASE_PATH, './api')
+      const API_REST_PORT = 9010
+      const API_WS_DIR_SRC = path.resolve(BASE_PATH, './api')
+      const API_WS_PORT = 9091
+      const WEBCHAT_DIR_SRC = path.resolve(BASE_PATH, './webchat')
+      const WEBCHAT_PORT = 9000
+
+      const pgs = new PlaygroundService(API_REST_PORT)
+      try {
+        await pgs.start()
+      } catch (e) {
+        console.log("Couldn't start the Playground service: ", e)
+      }
+
+      if (apps.includes('websocket')) {
+        try {
+          await kill(API_WS_PORT, 'tcp')
+        } catch (e) {
+          console.error(
+            `Error whilst shutting down "api websocket" port: ${colors.red(
+              e.message
+            )}`
+          )
+        }
+      }
+
+      if (apps.includes('rest')) {
+        try {
+          await kill(API_REST_PORT, 'tcp')
+        } catch (e) {
+          console.error(
+            `Error whilst shutting down "api rest" port: ${colors.red(
+              e.message
+            )}`
+          )
+        }
+      }
+
+      if (apps.includes('webchat')) {
+        try {
+          await kill(WEBCHAT_PORT, 'tcp')
+        } catch (e) {
+          console.error(
+            `Error whilst shutting down "webchat" port: ${colors.red(
+              e.message
+            )}`
+          )
+        }
+      }
+
+      const jobs = {
+        webchat: {
+          name: 'webchat',
+          command: `yarn workspace webchat start --env playgroundCode=${pgs.code}`,
+          prefixColor: 'yellowBright.bgBlack',
+          runWhen: () => fs.existsSync(WEBCHAT_DIR_SRC),
+        },
+        rest: {
+          name: 'api.rest',
+          command: 'yarn workspace api start:rest',
+          prefixColor: 'blueBright.bgBlack',
+          runWhen: () => fs.existsSync(API_REST_DIR_SRC),
+        },
+        websocket: {
+          name: 'api.ws',
+          command: 'yarn workspace api start:websocket',
+          prefixColor: 'magentaBright.bgBlack',
+          runWhen: () => fs.existsSync(API_WS_DIR_SRC),
+        },
+      }
+
+      try {
+        await concurrently(
+          Object.keys(jobs)
+            .map(n => apps.includes(n) && jobs[n])
+            .filter(job => job && job.runWhen()),
+          {
+            //prefix: '{name} |',
+            timestampFormat: 'HH:mm:ss',
+          }
+        )
+      } catch (e) {
+        console.log(e)
+      }
+      await pgs.stop()
```

Comment:
What if `pgs.start()` call failed? Would this call fail and Botonic wouldn't be served?
If this is true, maybe this `pgs.stop()` call should be done only if `pgs.start()` was successful or within a `try/catch`.

Reply:
If it failed then this call won't do anything